broker/src/test/java/io/camunda/zeebe/broker/exporter/stream/ExternalExporterContainerTest.java

```diff
@@ -0,0 +1,203 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.broker.exporter.stream;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatCode;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import io.camunda.zeebe.broker.exporter.jar.ExporterJarClassLoader;
+import io.camunda.zeebe.broker.exporter.repo.ExporterLoadException;
+import io.camunda.zeebe.engine.processing.streamprocessor.TypedRecord;
+import io.camunda.zeebe.exporter.api.Exporter;
+import io.camunda.zeebe.exporter.api.context.Context;
+import io.camunda.zeebe.exporter.api.context.Controller;
+import io.camunda.zeebe.protocol.impl.record.RecordMetadata;
+import io.camunda.zeebe.protocol.record.Record;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+import net.bytebuddy.ByteBuddy;
+import net.bytebuddy.dynamic.DynamicType.Unloaded;
+import org.agrona.CloseHelper;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+import org.junit.jupiter.api.parallel.Execution;
+import org.junit.jupiter.api.parallel.ExecutionMode;
+
+/**
+ * This suite tests external exporters, that is, those loaded via a self-contained JAR. For thread
+ * context class loader (TCL) tests, it will create a JAR which contains the following: a subclass
+ * of {@link TclExporter} called {@code com.acme.TestExporter}, which captures the TCL for every
+ * method called, and a simple empty class called {@code com.acme.Config}.
+ *
+ * <p>To verify that the TCL is correctly set, we compare the captured TCL for each method with the
+ * exporter class' class loader. When an external exporter is loaded, a new {@link
+ * ExporterJarClassLoader} instance is created for it, and the exporter class has its class loader
+ * property set to it. We can then compare the captured TCL with that.
+ */
+@Execution(ExecutionMode.CONCURRENT)
+final class ExternalExporterContainerTest {
+  private static final String EXPORTER_CLASS_NAME = "com.acme.TestExporter";
+  private ExporterContainerRuntime runtime;
+
+  @BeforeEach
+  void beforeEach(final @TempDir Path storagePath) {
+    runtime = new ExporterContainerRuntime(storagePath);
```

Comment:
I'm not that familiar with junit5, but the `@Execution` is set to `CONCURRENT`. Wouldn't that mean the runtime is overwritten concurrently? Or does each test get its own instance of this test class?

Reply:
No, the lifecycle is still as you expect - for each test, an instance of the class is created, so instance-level properties are isolated.